sale_multicompany_reporting_currency/models/res_config_settings.py

```diff
@@ -0,0 +1,21 @@
+# Copyright 2022 Camptocamp SA
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html)
+
+from odoo import fields, models
+
+
+class ResConfigSettings(models.TransientModel):
+    _inherit = "res.config.settings"
+
+    amount_option = fields.Selection(related="company_id.amount_option", readonly=False)
+
+    def set_values(self):
+        sale_order = self.env["sale.order"]
+        applied_currency = sale_order._get_multicompany_reporting_currency_id()
+        super().set_values()
+        to_apply_currency = self.multicompany_reporting_currency
+        if applied_currency.id != to_apply_currency.id:
+            sale_order.search([]).write(
+                {"multicompany_reporting_currency_id": to_apply_currency.id}
+            )
+        return True
```

Comment:
@vvrossem thanks for your work ;)

IMO we need to update the ``amount_multicompany_reporting_currency`` field on the SO's every time we change the settings, to have the right displayed amount.

```suggestion
    def set_values(self):
        sale_order = self.env["sale.order"]
        applied_currency = sale_order._get_multicompany_reporting_currency_id()
        super().set_values()
        to_apply_currency = self.multicompany_reporting_currency
        sales = sale_order.search([])
        if applied_currency.id != to_apply_currency.id:
            sales.write(
                {"multicompany_reporting_currency_id": to_apply_currency.id}
            )
        else:
            sales._compute_amount_multicompany_reporting_currency()
        return True
```

Reply:
@imlopes thank you for noticing this behavior. It's fixed: https://github.com/OCA/sale-reporting/pull/262/commits/1d9b72d26c16e926e84f134eda0c584064c80840

Reply:
do we really need a dependency on `company_id.amount_option`?
When you change a system setting `set_values` is called and normally with 
```
sales.write({"multicompany_reporting_currency_id": to_apply_currency.id})
```
existing orders with different currencies will be recomputed.
I don't think we need either `sales._compute_amount_multicompany_reporting_currency()`, because it's additional recomputation on saving settings. But, redundant as we recompute **orders in EURO to orders in EURO**. Why do we need that additional recompute which will impact the performance?

Reply:
Without `@api.depends("company_id.amount_option", (...)`, `amount_multicompany_reporting_currency` is not recomputed when changing `amount_option` in the system settings